src/internal/components/options-list/utils/use-highlight-option.ts

```diff
@@ -65,12 +67,13 @@ export function useHighlightedOption<OptionType>({
   return [
     { highlightType, highlightedIndex, highlightedOption },
```

Comment:
Since we already return an object here (some sort of `highlightedState`) can we include the new field on the top level rather than introducing a nested object

Reply:
the existing object properties get passed around to other components as individual properties not the full object, so it feels to me cleaner to pass it around in this way rather than creating an additional property on many subcomponents